src/CodeGenerators/SettingsGen/FileGeneration/XmlFileGeneration.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Serialization;
+
+namespace Microsoft.Omex.CodeGenerators.SettingsGen.FileGeneration
+{
+	/// <summary>
+	/// Generates an xml file with the settings model
+	/// </summary>
+	/// <typeparam name="TSettings">Settings model type</typeparam>
+	public class XmlFileGeneration<TSettings> : IFileGenerator<TSettings> where TSettings : class
+	{
+		/// <inheritdoc/>
+		public string FileType => "XML";
+
+		/// <inheritdoc />
+		public void GenerateFile(TSettings settings, string filename)
+		{
+			string content = XmlFileContents(settings);
+
+			// Serializing the xml file creates a different format of the second line in the file as Service Fabric.
+			// SF changes the header when the application project gets build and creates changes to the file if these are not swapped.
+			content = content.Replace("xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\"", "xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"");
```

Comment:
Could the same be achieved by `XmlTextWritter.WriteAttributeString` or something similar to avoid dumping everything to string?

Reply:
This was able to removed